### PR TITLE
Add request envelope to support Multiserving frameworks

### PR DESCRIPTION
**Pull request by abishekchiffon**
## Description

Adds the wrapper feature to help Multiserving frameworks.
The envelope parses request from multiple Model serving frameworks like Seldon, KFServing, without any modifications in the handler code.
This PR builds on top of the PR #521  from [Htappen](https://github.com/htappen/serve/tree/issue_411)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Feature/Issue validation/testing

-Additional units added to ts/torch_handler/unit_tests. 
-Executed end to end flow and verified following
 Gradle build logs :[build_gradle_411_envelope.log](https://github.com/pytorch/serve/files/5473016/build_gradle_411_envelope.log)
Unit test logs: [unit_tests_411_envelope.log](https://github.com/pytorch/serve/files/5473022/unit_tests_411_envelope.log)
Regression suite logs: [regression_test_411_envelope.log](https://github.com/pytorch/serve/files/5473029/regression_test_411_envelope.log)

## Checklist:

- [x] Have you added tests that prove your fix is effective or that this feature works?
- [x] New and existing unit tests pass locally with these changes?
- [x] Has code been commented, particularly in hard-to-understand areas?
- [x] Have you made corresponding changes to the documentation?

### CODE REVIEW

**Comment by chauhang** [comment]
@abishekchiffon Thanks for getting this PR ready. Can you please attach the detailed testing logs as well?

**Comment by maaquib** [comment]
@abishekchiffon Seeing failures when running the unit test `python -m pytest test_envelopes.py`

```
========================================================================= short test summary info ==========================================================================
ERROR test_envelopes.py::test_json - RuntimeError: Missing the model.pt file
ERROR test_envelopes.py::test_json_batch - RuntimeError: Missing the model.pt file
ERROR test_envelopes.py::test_json_double_batch - RuntimeError: Missing the model.pt file
ERROR test_envelopes.py::test_body - RuntimeError: Missing the model.pt file
======================================================================= 1 passed, 4 errors in 0.55s ========================================================================
```

**Review by maaquib — CHANGES_REQUESTED** [review]
@abishekchiffon Seeing failures when running the unit test `python -m pytest test_envelopes.py`

```
========================================================================= short test summary info ==========================================================================
ERROR test_envelopes.py::test_json - RuntimeError: Missing the model.pt file
ERROR test_envelopes.py::test_json_batch - RuntimeError: Missing the model.pt file
ERROR test_envelopes.py::test_json_double_batch - RuntimeError: Missing the model.pt file
ERROR test_envelopes.py::test_body - RuntimeError: Missing the model.pt file
======================================================================= 1 passed, 4 errors in 0.55s ========================================================================
```

File: frontend/server/src/main/java/org/pytorch/serve/util/codec/ModelRequestEncoder.java [hunk 1]
```diff
@@ -43,10 +43,23 @@ protected void encode(ChannelHandlerContext ctx, BaseModelRequest msg, ByteBuf o
                 buf = handler.getBytes(StandardCharsets.UTF_8);
             }
 
+            // TODO: this might be a bug. If handler isn't specified, this
```
[the remark this answers is not in the document]
Comment (abishekchiffon) [reply frontend/server/src/main/java/org/pytorch/serve/util/codec/ModelRequestEncoder.java:46]: I will create an issue for the bug.

File: docs/default_handlers.md [hunk 2]
```diff
@@ -46,4 +46,7 @@ For more details see [examples](https://github.com/pytorch/serve/tree/master/exa
 - [object_detector](https://github.com/pytorch/serve/tree/master/examples/object_detector/index_to_name.json)
 
 # Contributing
-If you'd like to edit or create a new default_handler class, make sure to run and update the unit tests in [unit_tests](https://github.com/pytorch/serve/tree/master/ts/torch_handler/unit_tests). As always, make sure to run [torchserve_sanity.sh](https://github.com/pytorch/serve/tree/master/torchserve_sanity.sh) before submitting.
+If you'd like to edit or create a new default_handler class, you need to take the following steps:
+1. Write a new class derived from BaseHandler. Add it as a separate file in `ts/torch_handler/`
+1. Update `model-archiver/model_packaging.py` to add in your classes name
+1. Run and update the unit tests in [unit_tests](https://github.com/pytorch/serve/tree/master/ts/torch_handler/unit_tests). As always, make sure to run [torchserve_sanity.sh](https://github.com/pytorch/serve/tree/master/torchserve_sanity.sh) before submitting.
```
[the remark this answers is not in the document]
Comment (abishekchiffon) [reply docs/default_handlers.md]: **The test suite can be either ran by the   `./ts/torch_handler/unit_tests/run_unit_tests.sh`** command or by running
`python -m pytest test_envelopes.py` after the model artifacts are moved to the _ts/torch_handler/unit_tests/models/tmp_ folder.
As all the unit test files requires the model artifacts to be present inside _'tmp'_ folder. These model artifacts are downloaded or moved to the _'tmp'_ in '[ts/torch_handler/unit_tests/run_unit_tests.sh](https://github.com/abishekchiffon/serve/blob/issue_411_envelope/ts/torch_handler/unit_tests/run_unit_tests.sh)' and deleted on completion.

[the remark this answers is not in the document]
Comment (abishekchiffon) [reply docs/default_handlers.md]: > Made the suggested change in the default_handlers.

[the remark this answers is not in the document]
Comment (abishekchiffon) [reply docs/default_handlers.md]: Made the suggested change in the default_handlers.

**Review by chauhang — APPROVED** [review]

**Review by dhanainme — APPROVED** [review]

**Pull request merged**